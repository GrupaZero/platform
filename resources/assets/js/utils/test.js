import Translations from '../lang/locales.js';
import VueI18n from 'vue-i18n';
import {mount} from 'avoriaz';

function vueMount(component, arg) {
    component.i18n = new VueI18n({
        locale: 'en',
        fallbackLocale: 'en',
        messages: Translations
    });
    return mount(component, arg)
}

export {vueMount};