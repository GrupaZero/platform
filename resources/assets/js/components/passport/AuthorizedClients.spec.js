import test from 'ava'
import {vueMount} from '../../utils/test'
import Bluebird from 'bluebird'
import AuthorizedClients from './AuthorizedClients.vue'

test.beforeEach(() => {
    axiosMock.reset()
    axiosMock.delayResponse = 0
})

test.serial('Init state without tokens', t => {
    t.plan(1)

    axiosMock.onGet('/oauth/tokens').reply(200, [])

    const wrapper = vueMount(AuthorizedClients)

    return Bluebird
        .delay()
        .then(() => {
            t.snapshot(wrapper.html())
        })

})

test.serial('Should render OAuth tokens table', t => {
    t.plan(3)

    axiosMock.onGet('/oauth/tokens')
        .reply(200, [
            {id: 1, client: {name: 'OAuth Token 1'}, scopes: ['scope1', 'scope2']},
            {id: 2, client: {name: 'OAuth Token 2'}, scopes: ['scope1', 'scope3']}
        ])

    const wrapper = vueMount(AuthorizedClients)
    t.is(0, wrapper.find('tr').length) // Before rendering

    return Bluebird
        .delay()
        .then(() => {
            t.snapshot(wrapper.html())
            t.is(3, wrapper.find('tr').length) // 2 tokens + header
        })

})

test.serial('Should render OAuth tokens table after 100 ms delay', t => {
    t.plan(3)

    axiosMock.delayResponse = 100

    axiosMock.onGet('/oauth/tokens')
        .reply(200, [
            {id: 1, client: {name: 'OAuth Token 1'}, scopes: ['scope1', 'scope2']},
            {id: 2, client: {name: 'OAuth Token 2'}, scopes: ['scope1', 'scope3']}
        ])

    const wrapper = vueMount(AuthorizedClients)
    t.is(0, wrapper.find('tr').length) // Before rendering

    return Bluebird
        .delay(105)
        .then(() => {
            t.snapshot(wrapper.html())
            t.is(3, wrapper.find('tr').length) // 2 tokens + header
        })

})
