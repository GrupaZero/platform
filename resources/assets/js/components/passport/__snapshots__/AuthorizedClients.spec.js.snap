// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Init state without tokens 1`] = `"<div><!----></div>"`;

exports[`Should render OAuth tokens table 1`] = `
"<div><div><div class=\\"panel panel-default\\"><div class=\\"panel-heading\\">Authorized Applications</div><div class=\\"panel-body\\"><table class=\\"table table-borderless m-b-none\\"><thead><tr><th>Name</th><th>Scopes</th><th></th></tr></thead><tbody><tr><td style=\\"vertical-align: middle;\\">
                                    OAuth Token 1
                                </td><td style=\\"vertical-align: middle;\\"><span>
                                        scope1, scope2
                                    </span></td><td style=\\"vertical-align: middle;\\"><a class=\\"action-link text-danger\\">
                                        Revoke
                                    </a></td></tr><tr><td style=\\"vertical-align: middle;\\">
                                    OAuth Token 2
                                </td><td style=\\"vertical-align: middle;\\"><span>
                                        scope1, scope3
                                    </span></td><td style=\\"vertical-align: middle;\\"><a class=\\"action-link text-danger\\">
                                        Revoke
                                    </a></td></tr></tbody></table></div></div></div></div>"
`;

exports[`Should render OAuth tokens table after 100 ms delay 1`] = `
"<div><div><div class=\\"panel panel-default\\"><div class=\\"panel-heading\\">Authorized Applications</div><div class=\\"panel-body\\"><table class=\\"table table-borderless m-b-none\\"><thead><tr><th>Name</th><th>Scopes</th><th></th></tr></thead><tbody><tr><td style=\\"vertical-align: middle;\\">
                                    OAuth Token 1
                                </td><td style=\\"vertical-align: middle;\\"><span>
                                        scope1, scope2
                                    </span></td><td style=\\"vertical-align: middle;\\"><a class=\\"action-link text-danger\\">
                                        Revoke
                                    </a></td></tr><tr><td style=\\"vertical-align: middle;\\">
                                    OAuth Token 2
                                </td><td style=\\"vertical-align: middle;\\"><span>
                                        scope1, scope3
                                    </span></td><td style=\\"vertical-align: middle;\\"><a class=\\"action-link text-danger\\">
                                        Revoke
                                    </a></td></tr></tbody></table></div></div></div></div>"
`;
